airbyte-integrations/connectors/destination-databricks/src/main/java/io/airbyte/integrations/destination/databricks/DatabricksAzureBlobStorageStreamCopier.java

```diff
@@ -172,9 +180,18 @@ public String generateMergeStatement(final String destTableName) {
     LOGGER.info("Preparing to merge tmp table {} to dest table: {}, schema: {}, in destination.", tmpTableName, destTableName, schemaName);
     final var queries = new StringBuilder();
     if (destinationSyncMode.equals(DestinationSyncMode.OVERWRITE)) {
-      queries.append(sqlOperations.truncateTableQuery(database, schemaName, destTableName));
+      if (!useMetastore) {
+        queries.append(sqlOperations.truncateTableQuery(database, String.format("%s.%s", catalogName, schemaName), destTableName));
+      } else {
+        queries.append(sqlOperations.truncateTableQuery(database, schemaName, destTableName));
+      }
       LOGGER.info("Destination OVERWRITE mode detected. Dest table: {}, schema: {}, truncated.", destTableName, schemaName);
     }
+    if (!useMetastore) {
+        queries.append(sqlOperations.copyTableQuery(database, String.format("%s.%s", catalogName, schemaName), tmpTableName, destTableName));
+    } else {
+      queries.append(sqlOperations.copyTableQuery(database, schemaName, tmpTableName, destTableName));
+    }
     queries.append(sqlOperations.insertTableQuery(database, schemaName, tmpTableName, destTableName));
```

Comment:
I believe there should be `insertTableQuery` instead of `copyTableQuery`.

Reply:
@IgSaf you were right, I fixed the call and remove the unnecessary one in line 195